l3experimental/l3opacity/l3opacity.dtx

```diff
@@ -117,14 +125,13 @@
 \cs_new_protected:Npn \opacity_stroke:n #1
   { \@@_select:nN {#1} \@@_backend_stroke:n }
 \cs_new_protected:Npn \@@_select:nN #1#2
-  { \exp_args:Ne \@@_select_aux:nN { \fp_eval:n {#1} } #2 }
-\cs_new_protected:Npn \@@_select_aux:nN #1#2
   {
+    \fp_set:Nn \l_@@_tmpa_fp { #1 }
     \bool_lazy_or:nnTF
-      { \fp_compare_p:nNn {#1} < { 0 } }
-      { \fp_compare_p:nNn {#1} > { 1 } }
+      { \fp_compare_p:nNn { \l_@@_tmpa_fp } < { \c_zero_fp } }
+      { \fp_compare_p:nNn { \l_@@_tmpa_fp } > { \c_one_fp } }
```

Comment:
Drop the braces here - we allow them to be elided. Personally I don't particularly like using `\c_..._fp` other than in the tightest of loops, but I can live with it here.